docs/Configuration/agent-configuration.md

```diff
@@ -272,6 +272,53 @@ In the above example:
 - the `hello_world_macos` extension is deployed to macOS hosts that are members of the 'Zoom installed' label.
 - the `hello_world_linux` extension is deployed to Linux hosts that are members of the 'Ubuntu Linux' **and** 'Zoom installed' labels.
 
+### Configure fleetd update channels
+
+_Available in Fleet Premium v4.43.0 and fleetd v1.20.0_
+
+Users can configure fleetd TUF auto-update channels for its components from Fleet's agent options. The components that can be configured are `orbit`, `osqueryd` and `desktop` (Fleet Desktop). When one of these components is omitted in `update_channels` then `stable` is assumed as the value for such component.
+
+Examples:
+```yaml
+apiVersion: v1
+kind: config
+spec:
+  agent_options:
+    update_channels: # requires Fleet's osquery installer
+      orbit: stable
+      osqueryd: '5.10.2'
+      desktop: edge
+```
+```yaml
+apiVersion: v1
+kind: config
+spec:
+  agent_options:
+    update_channels: # requires Fleet's osquery installer
+      orbit: edge
+      osqueryd: '5.10.2'
+      # in this configuration `desktop` is assumed to be "stable"
+```
+
+- If a configured channel doesn't exist in the TUF repository, then fleetd will log errors on the hosts and will not auto-update the component/s until the channel is changed to a valid value in Fleet's `update_channels` configuration or until the user pushes the component to the channel (which effectively creates the channel).
+- If the `update_channels` setting is removed from the agent settings, the devices will continue to use the latest configured channels.
+- If Fleet Desktop is disabled in fleetd, then the `desktop` channel setting is ignored by the host.
+
+#### Auto update startup loop
+
+Following we document an edge case scenario that could happen when using this feature.
+
+After upgrading `orbit` on your devices to `1.20.0` using this feature, beware of downgrading `orbit` by changing it to a channel that's older than `1.20.0`. The auto-update system in orbit could end up in an update startup loop (where orbit starts, changes its channel and restarts over and over).
+
+Following are the conditions (to avoid) that lead to the auto-update loop:
+1. fleetd with `orbit` < `1.20.0` was packaged/configured to run with orbit channel `A`.
+2. `orbit`'s channel `A` is updated to >= `1.20.0`.
+3. `orbit`'s channel in the Fleet agent settings is configured to `B`, where channel `B` has orbit version < `1.20.0`.
+
+This update startup loop can be fixed by any one of these actions:
+A. Downgrading channel `A` to < `1.20.0`.
+B. Upgrading channel `B` to >= `1.20.0`.
+
```

Comment:
What do you mean by "issue comment"?

Reply:
I mean put a comment in https://github.com/fleetdm/fleet/issues/13825

Reply:
Am all ears. @noahtalerman please let us know what's a better place to put this documentation of the edge case (and how to solve it).

Reply:
Moving the docs out of the PR to avoid blocking the feature.

Reply:
@lucasmrod can we prevent downgrades entirely? 

I'm thinking if fleetd finds that if either the osquery and Orbit versions set remotely (in Fleet YAML) are lower than it's current versions, the upgrade fails and fleetd logs a clear error message.

Also, we clearly document that downgrading Orbit and osquery isn't supported.

Why? I don't see any reason to support a downgrade workflow remotely. To avoid issues during upgrade, users will use the best practice rollout approach in which they upgrade the canary team, see if everything works as expected, and then upgrade the production team.

In addition, current and future features rely on specific Orbit and osquery versions (I'm thinking remote agent options and some host vitals). If we allow downgrades, users might accidentally shoot themselves in the foot and break these features.

If we're wrong and need to support downgrades later I don't why we couldn't add it. I doubt we will tho.